src/js/program/background-media.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * A simple data structure for containing both of the background loading objects.
+ * loadedMedia is the currently active item which has been put into the background during ad playback.
+ * loadingMedia is an item which is preloading in the background which may be selected in the future. It is usually the
```

Comment:
Wouldn't `nextItem`, `nextPromise` and `currentMedia` be clearer? I feel like I'm going to be doing a lot of double-takes outside of this file with loaded/loading and Media where it's a mediaController vs a Promise.